torch/fx/experimental/sym_node.py

```diff
@@ -1053,6 +1055,19 @@ def is_constant(x):
             return x.node.is_constant()
         return False
 
+    if method in bool_becomes_int_magic_methods:
+
+        def promote(x):
+            """Implements True+True=2, which works in python but not sympy"""
+            if isinstance(x, SymBool):
+                return SymInt(x.node.wrap_int(int(x)))
```

Comment:
calling `int(x)` guards on the value of `x`. It would be better to implement a casting method similar to `sym_float` that casts a bool to an int, similar to the method `sym_float`.

Reply:
1) SymPy doesn't seem to support this
2) All the usage of this pattern I found are immediately fed into conditionals, so we need to guard on it anyway

Reply:
Regarding 1, we can work around it by using `Where` https://github.com/pytorch/pytorch/blob/e6b3a8ce5ff543a6a96044604b7e39497ba756a6/torch/utils/_sympy/functions.py#L151-L163.

Regarding 2, fair enough, although it's not an extremely unusual pattern to use `True` as 1 when doing things like modular computations. May be good to leave a comment.

Reply:
Since we guard on 0/1, wouldn't something like `sym_float` end up guarding anyway ?

Reply:
ah, good point :D